### DISCUSSION TITLE
Config with default of None requires type annotation unnecessarily

### DISCUSSION THREAD

**Discussion by chdsbd** [encode/starlette#1730] [Potential Issue] [answered]
I think we could add another overload to `starlette.config.Config` to support typing this following example

```
from starlette.config import Config

config = Config()
PRIVATE_KEY = config("PRIVATE_KEY", default=None)
reveal_type(PRIVATE_KEY) # Type of "GITHUB_PRIVATE_KEY" is "Any"
```

An extra overload could correctly type this variable as `str | None`

Would you accept a PR to fix this?

**Comment by Kludex** [comment, marked as answer]
hmmm... Isn't the above this case: https://github.com/encode/starlette/blob/25a52fe52a1fa4f55da4096fb0c7e6f6de0ae980/starlette/config.py#L84-L88 ?

> Would you accept a PR to fix this?

But yes...
